src/components/ProductList/ProductList.component.jsx

```diff
@@ -7,19 +7,24 @@ import useFiltering from './useFiltering';
 import useProductCategories from 'hooks/useProductCategories';
 import useProducts from 'hooks/useProducts';
 
-export const ProductListComponent = ({className, categories}) => {
+import {useQuery} from 'hooks/useQuery';
+
+export const ProductListComponent = ({className}) => {
+    let query = useQuery();
+    const queryString = query.get("category");
+
     const [productCategories] = useProductCategories();
-    const [products] = useProducts();
+    const [products, pageNumber, setPageNumber] = useProducts({page: 1, pageSize: 12});
+    const [filteredProducts, filters, handleCustomFiltering, clearAllFilters] = useFiltering(products, queryString);
 
-    const [filteredProducts, filters, handleCustomFilering] = useFiltering(products);
-   
     return (
         <div className={className}>
             <Sidebar data={productCategories} 
-                     clickEvent={handleCustomFilering} 
+                     clickEvent={handleCustomFiltering} 
+                     clearAllFilters={clearAllFilters}
                      title={`Categories`} 
                      filters={filters} />
-            <Grid data={filteredProducts} />
+            <Grid data={{ page: pageNumber, totalPages: products.totalPages, parsedData: filteredProducts, isLoading: products.isLoading}} pagination setPageNumber={setPageNumber} />
```

Comment:
I think that if you create an object for this would be better to understand 
```suggestion
            const gridData = { page: pageNumber, totalPages: products.totalPages, parsedData: filteredProducts, isLoading: products.isLoading };
            <Grid data={gridData} pagination setPageNumber={setPageNumber} />
```